go/extractor/project/project.go

```diff
@@ -477,6 +478,16 @@ func getBuildRoots(emitDiagnostics bool) (goWorkspaces []GoWorkspace, totalModul
 	return
 }
 
+// Determines whether `str` starts with any of `prefixes`.
+func startsWithAnyOf(str string, prefixes []string) bool {
+	for _, prefix := range prefixes {
+		if strings.HasPrefix(str, prefix) {
```

Comment:
Watch out for the usual pitfall ref string prefixes vs. path prefixes (is `a/b` a prefix of `a/bb`?)

Reply:
I agree this is an issue. The `filepath` package has a function `hasPrefix` which is deprecated for precisely this reason. There doesn't seem to be a replacement that does what you'd want. Copilot gave me this:
```
func isParent(parent, child string) (bool, error) {
    rel, err := filepath.Rel(parent, child)
    if err != nil {
        return false, err
    }
    return !strings.Contains(rel, ".."), nil
}
```

Reply:
I implemented something similar and added a test in https://github.com/github/codeql/pull/15810/commits/ac394dc80ce6e3488285fd5abc01f2c5543ba0fb